### PR TITLE
test agent error reporting

### LINKED ISSUE #30: Add Tests for Errors
Currently only transactions are sent and checked

### PR DESCRIPTION
**Pull request by graphaelli**
Documents current integration testing application API and adds a new endpoint, `/oof`, that should trigger auto-instrumented error capturing.

closes #30

### CODE REVIEW

File: docker/nodejs/express/app.js [hunk 1]
```diff
@@ -48,6 +48,15 @@ function extra_route () {
     span.end()
 }
 
+app.use(function (err, req, res, next) {
+  console.error(err.stack)
+  res.status(500).send('Something broke!')
+});
```
Comment (Qard) [inline docker/nodejs/express/app.js]: Put this after the `/oof` route and it should work.

File: tests/agent/test_python.py [hunk 2]
```diff
@@ -7,8 +7,13 @@
...
 def test_req_flask(flask):
-    utils.check_agent_transaction(
-        flask.foo, flask.apm_server.elasticsearch, ct=2)
+    utils.check_agent_transaction(flask.foo, flask.apm_server.elasticsearch)
+
+
+@pytest.mark.version
+@pytest.mark.flask
+def test_flask_error(flask):
+    utils.check_agent_error(flask.oof, flask.apm_server.elasticsearch, ct=2)
```
Comment (graphaelli) [inline tests/agent/test_python.py]: @beniwohli I was surprised this generated two documents

Comment (beniwohli) [reply tests/agent/test_python.py]: I haven't looked at the actual documents, but this is probably due to [`logging=True`](https://github.com/elastic/apm-integration-testing/pull/203/files#diff-b4c662e3253b95a7af5361efeee2512dR24), which automatically sets up logging using our `LoggingHandler`. So one error is from our exception handler, and the other one from the logging handler, at least if my guess is right.

Comment (graphaelli) [reply tests/agent/test_python.py]: Looks right based on the documents produced, good with that for the tests?

**Pull request merged**